### ISSUE TITLE
conda throws error if allow_other_channels setting is used

### ISSUE THREAD

**Issue by JeremyEnns** [conda/conda#2891]
The feature to lock down what channels your users are allowed to use stopped working
http://conda.pydata.org/docs/install/central.html#allow-other-channels-allow-other-channels

Reproduced this error in Windows 10 and OS X 10.11.5, if you use this setting in the systemwide .condarc file.  

```
$ cat /Users/jenns/anaconda/.condarc
allow_other_channels: False
channels:
  - defaults

```

```
$ conda info
Traceback (most recent call last):
  File "/Users/jenns/anaconda/bin/conda", line 6, in <module>
    sys.exit(main())
  File "/Users/jenns/anaconda/lib/python2.7/site-packages/conda/cli/main.py", line 61, in main
    from conda.cli import conda_argparse
  File "/Users/jenns/anaconda/lib/python2.7/site-packages/conda/cli/conda_argparse.py", line 15, in <module>
    from .common import add_parser_help
  File "/Users/jenns/anaconda/lib/python2.7/site-packages/conda/cli/common.py", line 12, in <module>
    from ..config import (envs_dirs, default_prefix, platform, update_dependencies,
  File "/Users/jenns/anaconda/lib/python2.7/site-packages/conda/config.py", line 331, in <module>
    allowed_channels = get_allowed_channels()
  File "/Users/jenns/anaconda/lib/python2.7/site-packages/conda/config.py", line 329, in get_allowed_channels
    return normalize_urls(base_urls)
  File "/Users/jenns/anaconda/lib/python2.7/site-packages/conda/config.py", line 253, in normalize_urls
    urls = get_rc_urls() + urls
  File "/Users/jenns/anaconda/lib/python2.7/site-packages/conda/config.py", line 197, in get_rc_urls
    if rc.get('channels') is None:
AttributeError: 'NoneType' object has no attribute 'get'
```

**Comment by JeremyEnns** [comment]
downgraded conda to 4.0.9 and the issue went away.

**Comment by mcg1969** [comment]
This occurs because when `allow_other_channels` is `False`, the call to `get_allowed_channels()` in `config.py` is trying to call `get_rc_urls()` before the local `~/.condarc` has been loaded.

This can be fixed by moving the line `allowed_channels = get_allowed_channels()` to the very end of `config.py`, after the `load_condarc` call. It can also be fixed by modifying `get_rc_urls` to return an empty list if `rc` is `None`. I'm leaning towards the latter.

**Comment by sebastian-luna-valero** [comment]
I had the same problem after updating conda. My .condarc file is:
```
channels:
- defaults
- https://conda.anaconda.org/r
- https://conda.anaconda.org/bioconda
```
As suggested by @mcg1969 , updating `get_rc_urls` as:
```
def get_rc_urls():
    if rc is None:
        return []
    if rc.get('channels') is None:
        return []
    if 'system' in rc['channels']:
        raise RuntimeError("system cannot be used in .condarc")
    return rc['channels']
```

works for me.

Many thanks!

**Comment by kalefranz** [comment]
fixed in #2896
